Ingress Filtering support - documentation update (#2010)
kyma-project/kyma-environment-broker: docs/user/04-50-ingress-filtering.md

# Enable Ingress Filtering

> [!NOTE]

> > The traffic filtering feature is only available for SAP internal customers as it is integrated with SAP's geo-blocking solution.
> 
> > [!NOTE]
> > The ingress filtering feature is available for the following plans: aws, gcp, azure.

Reviewer: ```suggestion
> The ingress filtering feature is only available to SAP internal customers as it is integrated with SAP's geo-blocking solution.
> The ingress filtering feature is available for the `aws`, `gcp`, and `azure` plans.
```

> Kyma Environment Broker allows you to either enable or disable an ingress filtering during SAP BTP, Kyma runtime provisioning and update operations.

Reviewer 2: In my opinion we should use `provisioning and updating` or `provision and update` to keep it consistent.

Author: This is how we call these operations as far as I know: provisioning, update.
See custom networking, custom administrators etc.

> Kyma Environment Broker allows you to either enable or disable an ingress filtering during SAP BTP, Kyma runtime provisioning and update operations.

Reviewer: ```suggestion
Kyma Environment Broker allows you to enable or disable ingress filtering during SAP BTP, Kyma runtime provisioning and update operations.
```

> To do so, specify the additional **ingressFiltering** parameter in the provisioning or update request.
> By default, the ingress filtering is disabled.

Reviewer: ```suggestion
By default, ingress filtering is disabled.
To enable it, set the additional **ingressFiltering** parameter to `true` in the provisioning or update request.
```

   --header "$AUTHORIZATION_HEADER" \
   --data-raw "{
       \"service_id\": \"47c9dcbf-ff30-448e-ab36-d3bad66ba281\",
       \"plan_id\": \"4deee563-e5ec-4731-b9b1-53b42d855f0c\",
       \"context\": {
           \"globalaccount_id\": \"$GLOBAL_ACCOUNT_ID\",
           \"subaccount_id\": \"$SUBACCOUNT_ID\",
           \"user_id\": \"$USER_ID\",
       },
       \"parameters\": {
           \"name\": \"$NAME\",
           \"region\": \"$REGION\",
           \"ingressFiltering\": true
       }
   }"
```

See the example of the update request:

```bash
   export VERSION=1.15.0
   curl --request PATCH "https://$BROKER_URL/oauth/v2/service_instances/$INSTANCE_ID?accepts_incomplete=true" \
   --header 'X-Broker-API-Version: 2.14' \
   --header 'Content-Type: application/json' \
   --header "$AUTHORIZATION_HEADER" \
   --data-raw "{
       \"service_id\": \"47c9dcbf-ff30-448e-ab36-d3bad66ba281\",
       \"plan_id\": \"4deee563-e5ec-4731-b9b1-53b42d855f0c\",
       \"context\": {
           \"globalaccount_id\": \"$GLOBAL_ACCOUNT_ID\",
           \"subaccount_id\": \"$SUBACCOUNT_ID\",
       },
       \"parameters\": {
           \"ingressFiltering\": true
       }
   }"
```

> [!NOTE]

> > Attempt to enable or disable ingress filtering for unsupported plan or external customer will result in an error.

Reviewer: ```suggestion
> Attempt to enable or disable ingress filtering for an unsupported plan or an external customer results in an error.
```
